### PR TITLE
Add constructors for Select

### LINKED ISSUE #5567: Some useful tree constructors
In implementing ScalaTest macros, I find two tree constructors are useful:

```Scala
def selectApply(receiver: Term, name: Name, args: List[Term]): Term
def let(expr: Term)(body: Ident => Term): Term
```

The constructor `selectApply` should do _overloading resolution_ internally, but no other adapation (like inserting `apply` nor implicits). The constructor `let` allows creating bindings friendly.

The current implementation of `assert` is as follows:

```Scala
      case Term.Apply(Term.Select(lhs, op), rhs :: Nil) =>
        op match {
          case "==" =>
            val left = lhs.seal[Any]
            val right = rhs.seal[Any]
            '{
              val _left   = ~left
              val _right  = ~right
              val _result = _left == _right
              val _bool = Bool.binaryMacroBool(_left, ~op.toExpr, _right, _result, ~prettifier)
              Assertions.assertionsHelper.macroAssert(_bool, ~clue, ~pos)
            }
          case ">" =>
            // blocked by tasty constructors
            // https://github.com/lampepfl/dotty/pull/5438
            val left = lhs.seal[Int]
            val right = rhs.seal[Int]
            '{
              val _left   = ~left
              val _right  = ~right
              val _result = _left > _right
              val _bool = Bool.binaryMacroBool(_left, ~op.toExpr, _right, _result, ~prettifier)
              Assertions.assertionsHelper.macroAssert(_bool, ~clue, ~pos)
            }
```

With the two constructors, the implementation can be made more generic & simpler:

```Scala
case Term.Apply(Term.Select(lhs, op), rhs :: Nil) =>
  val res =
  let(lhs) { left =>
    let(rhs) { right =>
      let(selectApply(left, op, right)) { result =>
         val l = left.seal[Any]
         val r = right.seal[Any]
         val b = result.seal[Boolean]
        '{
            val _bool = Bool.binaryMacroBool(~l, ~op.toExpr, ~r, ~b, ~prettifier)
            Assertions.assertionsHelper.macroAssert(_bool, ~clue, ~pos)      
         }
    }
  }
  res.seal[Assertion]
```

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #5567 to add contructors for Select

Add constructors for Select. The usage of `Select.copy` is possible sometimes instead of the constructors. However, `Select.copy` is not as expressive, verbose, the semantics is unspecified, and it  causes weird exceptions in ScalaTest.

- [x] decide what to do with positions for synthetic trees

### CODE REVIEW

**Comment by liufengyun** [comment]
@nicolasstucki It would be nice if we can get this in recently to unblock @cheeseng and make progress on ScalaTest.

File: compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala [hunk 1]
```diff
@@ -12,6 +12,7 @@ trait TreeOpsImpl extends scala.tasty.reflect.TreeOps with CoreImpl with Helpers
   def TreeDeco(tree: Tree): TreeAPI = new TreeAPI {
     def pos(implicit ctx: Context): Position = tree.sourcePos
     def symbol(implicit ctx: Context): Symbol = tree.symbol
+    def withPos(pos: Position)(implicit ctx: Context): Unit = tree.withSpan(pos.span)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala]: Try rebasing on top of #5940. This should fix the missing positions assertions.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala]: Thanks a lot. BTW, do we want to keep this method?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala]: I would remove it for now. We can add it later.

**Comment by liufengyun** [comment]
@nicolasstucki Can you give it another look? Thanks.

File: library/src/scala/tasty/reflect/TreeOps.scala [hunk 2]
```diff
@@ -272,8 +272,9 @@ trait TreeOps extends Core {
     /** Scala term selection */
     val Select: SelectModule
     abstract class SelectModule {
+      def unique(qualifier: Term, name: String)(implicit ctx: Context): Select
```
Comment (nicolasstucki) [inline library/src/scala/tasty/reflect/TreeOps.scala]: Add some documentation with the precondition that the method is not overloaded.

File: library/src/scala/tasty/reflect/TreeOps.scala [hunk 3]
```diff
@@ -272,8 +272,9 @@ trait TreeOps extends Core {
...
 
-      // TODO def apply(qualifier: Term, name: String, signature: Option[Signature])(implicit ctx: Context): Select
+      def overloaded(qualifier: Term, name: String, targs: List[Type], args: List[Term])(implicit ctx: Context): Apply
```
Comment (nicolasstucki) [inline library/src/scala/tasty/reflect/TreeOps.scala]: This should probably be the `apply` as it is the default constructor.

Comment (liufengyun) [reply library/src/scala/tasty/reflect/TreeOps.scala]: Indeed, `apply` will make the syntax shorter in usage. Here, I think there is one advantage of using `overloaded`: remind meta-programmers there is always possibility of overloading, thus the arguments should be provided for overloading resolution. That saves us from documenting the method and justifying the API design.

Comment (liufengyun) [reply library/src/scala/tasty/reflect/TreeOps.scala]: Another thought is that we may resort `apply` for selection with a symbol.

**Pull request merged**